powder_reduction.py

```diff
@@ -1,37 +1,142 @@
+        input_load_cal = {"InstrumentName": "WISH"}
+        cal = load_calibration(calibration, mantid_args=input_load_cal)
+        # Merge table with detector->spectrum mapping from sample
+        # (implicitly checking that detectors between sample and calibration are the same)
+        # cal = sc.merge(cal, sample.labels['detector_info'].value)
+        # Compute spectrum mask from detector mask
+        mask = sc.groupby(cal['mask'], group='spectrum', combine=Dim.Spectrum).any(Dim.Detector)
+
+        # Compute spectrum groups from detector groups
+        g = sc.groupby(cal['group'], group='spectrum', combine=Dim.Spectrum)
+
+        group = g.min(Dim.Detector)
+
+        assert group == g.max(Dim.Detector), \
+            "Calibration table has mismatching group for detectors in same spectrum"
+
+        sample.labels['group'] = group.data
+        sample.masks['mask'] = mask.data
+
+    # Correct 4th monitor spectrum
+    # There are 5 monitors for WISH. Only one, the fourth one, is selected for
+    # correction (like in the real WISH workflow).
+
+    # Select fourth monitor
+    mon4_selected = sample.attrs['monitor4'].values
+
+    # Smooth monitor
+    mon4_smooth = smooth_data.smooth_data(mon4_selected,
+                                          dim=Dim.Tof,
+                                          NPoints=40)
+    # Delete intermediate data
+    del mon4_selected
+
+    # Correct data
+    # 1. Normalize to monitor
+    # Convert to wavelength (counts and monitor)
+    sample_lambda = sc.neutron.convert(sample, Dim.Tof, Dim.Wavelength)
+
+    mon_conv = sc.neutron.convert(mon4_smooth, Dim.Tof, Dim.Wavelength)
+
+    # Rebin monitors' data
+    # The value 5615 corresponds to the value in Mantid after rebinning
+    lambda_min, lambda_max = LAMBDA_RANGE
+    mon_rebin = sc.rebin(mon_conv,
+                         Dim.Wavelength,
+                         sc.Variable([Dim.Wavelength],
+                                     unit=sc.units.angstrom,
+                                     values=np.linspace(lambda_min, lambda_max, num=5615)))
+    sample_lambda /= mon_rebin
+
+    # Delete intermediate data
+    del mon_rebin, mon_conv, sample
+
+    # Convert sample data back from wavelength to tof
+    sample_tof = sc.neutron.convert(sample_lambda, Dim.Wavelength, Dim.Tof)
+
+    # Delete intermediate data
+    del sample_lambda
+
+    # 2. Convert to d-spacing taking calibration into account
     if calibration is None:
         # No calibration data, use standard convert algorithm
-        dspacing = sc.neutron.convert(data, Dim.Tof, Dim.DSpacing)
+        sample_dspacing = sc.neutron.convert(data, Dim.Tof, Dim.DSpacing)
     else:
         # Calculate dspacing from calibration file
-        dspacing = sc.neutron.diffraction.convert_with_calibration(data,
-                                                                   calibration)
+        # Trick to avoid error in convert_with_calibration (bug)
+        del cal['mask']
```

Comment:
This is fixed now so these lines can be removed.

Reply:
I just checked with the last version of scipp. It works without deleting the mask in calibration.